apigee_edge.module

```diff
@@ -1371,6 +1372,15 @@ function apigee_edge_user_presave(UserInterface $account) {
           $logger->error("Unable to save @developer developer's developer id on user.", $context);
         }
       }
+      elseif ($previous->getEdgeErrorCode() === Developer::APIGEE_HYBRID_ERROR_CODE_DEVELOPER_EMAIL_MISMATCH) {
+        // Apigee X and Hybrid runtime v1.5.0 and v1.5.1 a call to change the developer's
+        // email address will not work so need to prevent user email update on
+        // Drupal as well.
+        // @see https://github.com/apigee/apigee-client-php/issues/153
+        // @see https://github.com/apigee/apigee-edge-drupal/issues/587
+        \Drupal::messenger()->addMessage("Unable to update developer's profile.", 'error');
```

Comment:
This hook can be triggered from various places (eg.: from update hooks, CLI commands, (JSON) API request, etc). Displaying a message with messenger only works for HTML requests/responses. Therefore calling a messenger here is not a good solution.
If a message should be displayed for this exception when an HTML response is expected then it should be done in an HTML exception subscriber.


This is one of the reasons why only logs are being created here.

An eye for one, I would not expose more information here for end-users, logs already contain enough information, end-user should just call customer support and report what happened.

Reply:
@mxr576
Removed the Drupal messenger and also updated the exception message, this is how it will look

![image](https://user-images.githubusercontent.com/75600200/123829614-3d909b00-d920-11eb-94a9-e42cd1563b9c.png)

@giteshk let me know if this looks good.